app/src/main/java/com/example/fittracker/activity/UserSettingsActivity.java

```diff
@@ -27,28 +27,27 @@ protected void onCreate(Bundle savedInstanceState) {
 
     public void init() {
         presenter = new UserSettingsPresenter(new UserSettingsView(this)
-                ,new UserSettingsModel(UserRoomDatabase.getDatabase(this))
-                ,getSharedPreferences(ConstantUtils.USER_PREFERENCES, MODE_PRIVATE)
-                .getInt(ConstantUtils.USER_PREFERENCES_ID,ConstantUtils.ZERO));
-        presenter.onCreate();
-    }//Check if its okay to pass the shared pref via constructor to presenter
+                ,new UserSettingsModel(UserRoomDatabase.getDatabase(this)
+                ,getSharedPreferences(ConstantUtils.USER_PREFERENCES, MODE_PRIVATE)));
+        presenter.init();
+    }
 
-    @OnClick(R.id.usersettings_activity_textview_email)
+    @OnClick(R.id.usersettings_activity_edittext_email)
     public void onEmailFocusChange() {
```

Comment:
if you need it.. you can put a boolean parameter so if you are focused will be true, and false if you are leaving the field.

public void onEmailFocusChange(boolean isFocused)

Reply:
All the focuschange views got replaced by normal onclick views.